flight/odlc_boundaries/search_path.py

```diff
@@ -0,0 +1,205 @@
+"""
+Functions for generating search paths to cover an area for finding the standard odlc objects
+"""
+
+from shapely.geometry import Polygon
+import utm
+from execute import move_to
+from mavsdk import System
+import logging
+import asyncio
+
+
+def latlon_to_utm(coords: dict[str, float | int | str]) -> dict[str, float | int | str]:
+    """
+    Converts latlon coordinates to utm coordinates and adds the data to the dictionary
+
+    Parameters
+    ----------
+    coords : Dict[str, float]
+        A dictionary containing lat long coordinates
+
+    Returns
+    -------
+    coords : Dict[str, float]
+        An updated dictionary with additional keys and values with utm data
+    """
+
+    utm_coords: tuple[float, float, int, str] = utm.from_latlon(
+        coords["latitude"], coords["longitude"]
+    )
+    coords["utm_x"] = utm_coords[0]
+    coords["utm_y"] = utm_coords[1]
+    coords["utm_zone_number"] = utm_coords[2]
+    coords["utm_zone_letter"] = utm_coords[3]
+    return coords
+
+
+def all_latlon_to_utm(
+    list_of_coords: list[dict[str, float | int | str]]
+) -> list[dict[str, float | int | str]]:
+    """
+    Converts a list of dictionaries with latlon data to add utm data
+
+    Parameters
+    ----------
+    list_of_coords : list[dict[str, union[float, int, str]]]
+        A list of dictionaries that contain lat long data
+
+    Returns
+    -------
+    list_of_coords : [dict[str, union[float, int, str]]]
+        An updated list of dictionaries with added utm data
+    """
+    coord: dict[str, float | int | str]
+    for coord, _ in enumerate(list_of_coords):
+        list_of_coords[coord] = latlon_to_utm(list_of_coords[coord])
+    return list_of_coords
+
+
+def generate_search_paths(
+    d_search_area_boundary: list[dict[str, float]], buffer_distance: int
+) -> list[tuple[float, float]]:
+    """
+    Generates a list of search paths of increasingly smaller sizes until the whole area
+    of the original shape has been covered
+
+    Parameters
+    ----------
+    d_search_area_boundary : List[Dict[str, float]]]
+        A list of coordinates in dictionary form that contain utm coordinate data
+    buffer_distance : int
+        The distance that each search path will be apart from the previous one.
+        For complete photographic coverage of the area, this should be equal to half the height
+        of the area the camera covers on the ground given the current altitude.
+
+    Returns
+    -------
+    generated_search_paths : [Tuple[float, float]]
+        A list of concentric search paths that cover the area of the polygon
+    """
+
+    # convert to shapely polygon for buffer operations
+    search_area_points: list[tuple[float, float]] = [
+        (point["utm_x"], point["utm_y"]) for point in d_search_area_boundary
+    ]
+    boundary_shape: Polygon = Polygon(search_area_points)
+
+    generated_search_paths: list[list[tuple[float, float]]]
```

Comment:
The errors for lines 95 and 99 in the pre-commit can be traced back here: this type annotation has one too many lists. The actual type for generated search paths is a list[tuple[float, float]]. So, its annotation should be:
generated_search_paths: list[tuple[float, float]]